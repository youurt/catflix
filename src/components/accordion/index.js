import React, { useState, useContext, createContext } from 'react';
import {
  Container,
  Inner,
  Title,
  Item,
  Header,
  Body,
  Frame,
} from './styles/accordion';

const ToggleContext = createContext();

const Accordion = ({ children, ...restProps }) => {
  return (
    <Container {...restProps}>
      <Inner>{children}</Inner>
    </Container>
  );
};
Accordion.Frame = function AccordionFrame({ children, ...restProps }) {
  return <Frame {...restProps}>{children}</Frame>;
};

Accordion.Title = ({ children, ...restProps }) => {
  return <Title {...restProps}>{children}</Title>;
};

Accordion.Item = ({ children, ...restProps }) => {
  const [toggleShow, setToggleShow] = useState(false);
  return (
    <ToggleContext.Provider value={{ toggleShow, setToggleShow }}>
      <Item {...restProps}>{children}</Item>
    </ToggleContext.Provider>
  );
};

Accordion.Header = ({ children, ...restProps }) => {
  const { toggleShow, setToggleShow } = useContext(ToggleContext);

  return (
    <Header
      onClick={() => setToggleShow((prevToggleShow) => !prevToggleShow)}
      {...restProps}
    >
      {children}
      {toggleShow ? (
        <img src="/images/icons/close-slim.png" alt="Close" />
      ) : (
        <img src="/images/icons/add.png" alt="Open" />
      )}
    </Header>
  );
};

Accordion.Body = ({ children, ...restProps }) => {
  const { toggleShow } = useContext(ToggleContext);

  return (
    <Body className={toggleShow ? 'open' : 'closed'} {...restProps}>
      <span>{children}</span>
    </Body>
  );
};

export default Accordion;
