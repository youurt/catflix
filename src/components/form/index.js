import React from 'react';
import {
  Container,
  Base,
  Error,
  Title,
  Text,
  TextSmall,
  Link,
  Input,
  Submit,
} from './styles/form';

const Form = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

Form.Base = ({ children, ...restProps }) => {
  return <Base {...restProps}>{children}</Base>;
};

Form.Error = ({ children, ...restProps }) => {
  return <Error {...restProps}>{children}</Error>;
};

Form.Title = ({ children, ...restProps }) => {
  return <Title {...restProps}>{children}</Title>;
};

Form.Text = ({ children, ...restProps }) => {
  return <Text {...restProps}>{children}</Text>;
};

Form.TextSmall = ({ children, ...restProps }) => {
  return <TextSmall {...restProps}>{children}</TextSmall>;
};

Form.Link = ({ children, ...restProps }) => {
  return <Link {...restProps}>{children}</Link>;
};

Form.Input = ({ children, ...restProps }) => {
  return <Input {...restProps}>{children}</Input>;
};

Form.Submit = ({ children, ...restProps }) => {
  return <Submit {...restProps}>{children}</Submit>;
};

export default Form;
