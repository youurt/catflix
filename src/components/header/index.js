import React from 'react';
import { Link as ReactRouterLink } from 'react-router-dom';
import { Background, Container, Logo, ButtonLink } from './styles/header';

const Header = ({ bg = true, children, ...restProps }) => {
  return bg ? <Background {...restProps}>{children}</Background> : children;
};

Header.Frame = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

Header.ButtonLink = ({ children, ...restProps }) => {
  return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

Header.Logo = ({ to, ...restProps }) => {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  );
};

export default Header;
