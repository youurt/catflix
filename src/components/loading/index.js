import React from 'react';
import { Spinner, LockBody, Picture, ReleaseBody } from './styles/loading';

const Loading = ({ src, ...restProps }) => {
  return (
    <Spinner {...restProps}>
      <LockBody />
      <Picture src={`/images/users/${src}.png`} data-testid="loading-picture" />
    </Spinner>
  );
};

Loading.ReleaseBody = () => {
  return <ReleaseBody />;
};

export default Loading;
