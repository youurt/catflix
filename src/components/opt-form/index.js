import React from 'react';
import { Container, Input, Button, Text, Break } from './styles/opt-form';

const OptForm = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

OptForm.Input = ({ ...restProps }) => {
  return <Input {...restProps} />;
};

OptForm.Button = ({ children, ...restProps }) => {
  return (
    <Button {...restProps}>
      {children}
      <img src="/images/icons/chevron-right.png" alt="Try Now!" />
    </Button>
  );
};

OptForm.Text = ({ children, ...restProps }) => {
  return <Text {...restProps}>{children}</Text>;
};

OptForm.Break = function OptFormBreak({ ...restProps }) {
  return <Break {...restProps} />;
};

export default OptForm;
