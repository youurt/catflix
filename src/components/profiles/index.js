import React from 'react';
import { Container, Title, List, Picture, Name, Item } from './styles/profiles';

const Profiles = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

Profiles.Title = ({ children, ...restProps }) => {
  return <Title {...restProps}>{children}</Title>;
};

Profiles.List = ({ children, ...restProps }) => {
  return <List {...restProps}>{children}</List>;
};

Profiles.User = ({ children, ...restProps }) => {
  return <Item {...restProps}>{children}</Item>;
};

Profiles.Name = ({ children, ...restProps }) => {
  return <Name {...restProps}>{children}</Name>;
};

Profiles.Picture = ({ src, ...restProps }) => {
  return (
    <Picture
      {...restProps}
      src={src ? `/images/users/${src}.png` : `/images/misc/loading.gif`}
    />
  );
};

export default Profiles;
