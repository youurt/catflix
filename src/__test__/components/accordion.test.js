import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import faqData from '../../fixtures/faqs.json';
import { Accordion } from '../../components';

describe('<Accordion/>', () => {
  it('should render the <Accordiotn/> with populated data', () => {
    const { container, getByText } = render(
      <Accordion>
        <Accordion.Title>Frequently Asked Questions</Accordion.Title>
        <Accordion.Frame>
          {faqData.map((item) => (
            <Accordion.Item key={item.id}>
              <Accordion.Header>{item.header}</Accordion.Header>
              <Accordion.Body>{item.body}</Accordion.Body>
            </Accordion.Item>
          ))}
        </Accordion.Frame>
      </Accordion>
    );

    expect(getByText('Frequently Asked Questions')).toBeTruthy();
    expect(getByText('What is Classicflix?')).toBeTruthy();
    expect(getByText('How much does Classicflix cost?')).toBeTruthy();
    expect(getByText('Where can I watch?')).toBeTruthy();
    expect(getByText('How do I cancel?')).toBeTruthy();
    expect(getByText('What can I watch on Classicflix?')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should open and close the <Accordion/> component', () => {
    const { container, queryByText, getByTestId } = render(
      <Accordion>
        <Accordion.Title>Frequently Asked Questions</Accordion.Title>
        <Accordion.Frame>
          {faqData.map((item) => (
            <Accordion.Item key={item.id}>
              <Accordion.Header data-testid="toggle_header">
                {item.header}
              </Accordion.Header>
              <Accordion.Body>{item.body}</Accordion.Body>
            </Accordion.Item>
          ))}
        </Accordion.Frame>
      </Accordion>
    );
    const whatIsClassicflixBodyText =
      'Watch Classicflix on your smartphone, tablet, smart TV, laptop or streaming device, all for one low fixed monthly fee. Plans start from 5.99 â‚¬ a month. No extra costs or contracts.';

    expect(queryByText('How much does Classicflix cost?')).toBeInTheDocument();
    fireEvent.click(queryByText('How much does Classicflix cost?'));
    // expect(queryByText(whatIsClassicflixBodyText)).toHaveClass('open');

    // expect(queryByText(whatIsClassicflixBodyText)).toBeTruthy();
    // fireEvent.click(queryByText('How much does Classicflix cost?'));
    // expect(queryByText(whatIsClassicflixBodyText)).toBeFalsy();

    // fireEvent.click(queryByText('How much does Classicflix cost?'));
    // expect(queryByText(whatIsClassicflixBodyText)).toBeFalsy();
    // expect(container.firstChild).toMatchSnapshot();
  });
});
