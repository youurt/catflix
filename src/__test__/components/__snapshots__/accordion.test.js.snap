// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion/> should render the <Accordiotn/> with populated data 1`] = `
<div
  class="accordion__Container-mofgfd-0 caNBTg"
>
  <div
    class="accordion__Inner-mofgfd-1 ffjCRz"
  >
    <h1
      class="accordion__Title-mofgfd-3 bXwGSX"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="accordion__Frame-mofgfd-2 humgWS"
    >
      <div
        class="accordion__Item-mofgfd-4 jnTzQP"
      >
        <div
          class="accordion__Header-mofgfd-5 kJikXx"
        >
          What is Classicflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
        <div
          class="accordion__Body-mofgfd-6 DXOIN closed"
        >
          <span>
            Classicflix is a streaming service that offers a wide variety of award-winning classic TV programmes and films – on thousands of internet-connected devices.

You can watch as much as you want, whenever you want, without a single advert – all for one low monthly price. There's always something new to discover, and new TV programmes and films are added every week!
          </span>
        </div>
      </div>
      <div
        class="accordion__Item-mofgfd-4 jnTzQP"
      >
        <div
          class="accordion__Header-mofgfd-5 kJikXx"
        >
          How much does Classicflix cost?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
        <div
          class="accordion__Body-mofgfd-6 DXOIN closed"
        >
          <span>
            Watch Classicflix on your smartphone, tablet, smart TV, laptop or streaming device, all for one low fixed monthly fee. Plans start from 5.99 € a month. No extra costs or contracts.
          </span>
        </div>
      </div>
      <div
        class="accordion__Item-mofgfd-4 jnTzQP"
      >
        <div
          class="accordion__Header-mofgfd-5 kJikXx"
        >
          Where can I watch?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
        <div
          class="accordion__Body-mofgfd-6 DXOIN closed"
        >
          <span>
            Watch anywhere, anytime, on an unlimited number of devices. Sign in with your Classicflix account to watch instantly on the web at classicflix.com from your personal computer or on any internet-connected device that offers the Classicflix app, including smart TVs, smartphones, tablets, streaming media players and game consoles.

You can also download your favourite programmes with the iOS, Android, or Windows 10 app. Use downloads to watch while you're on the go and without an internet connection. Take Classicflix with you anywhere.
          </span>
        </div>
      </div>
      <div
        class="accordion__Item-mofgfd-4 jnTzQP"
      >
        <div
          class="accordion__Header-mofgfd-5 kJikXx"
        >
          How do I cancel?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
        <div
          class="accordion__Body-mofgfd-6 DXOIN closed"
        >
          <span>
            Classicflix is flexible. There are no annoying contracts and no commitments. You can easily cancel your account online in two clicks. There are no cancellation fees – start or stop your account at any time.
          </span>
        </div>
      </div>
      <div
        class="accordion__Item-mofgfd-4 jnTzQP"
      >
        <div
          class="accordion__Header-mofgfd-5 kJikXx"
        >
          What can I watch on Classicflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
        <div
          class="accordion__Body-mofgfd-6 DXOIN closed"
        >
          <span>
            Classicflix has an extensive library of feature films, documentaries, TV programmes, anime, award-winning Classicflix originals, and more. Watch as much as you want, any time you want.
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
